src/eventgenerator/aggregator/aggregator.go

```diff
@@ -48,15 +51,31 @@ func (a *Aggregator) ConsumePolicy(policyMap map[string]*model.Policy, appChan c
 	if policyMap == nil {
 		return
 	}
+	var triggerArrayMap map[string][]*model.Trigger = make(map[string][]*model.Trigger)
 	for appId, policy := range policyMap {
 		for _, rule := range policy.TriggerRecord.ScalingRules {
 			appChan <- &model.AppMonitor{
 				AppId:      appId,
 				MetricType: rule.MetricType,
 				StatWindow: rule.StatWindow,
 			}
+			_, exist := triggerArrayMap[appId+"#"+rule.MetricType]
+			if !exist {
+				triggerArrayMap[appId+"#"+rule.MetricType] = []*model.Trigger{}
+			}
+			triggerArrayMap[appId+"#"+rule.MetricType] = append(triggerArrayMap[appId+"#"+rule.MetricType], &model.Trigger{
```

Comment:
This is wasteful.
You got it once and decided not to hold on to the object and then you get it again.
You should be able to do one retrieve and one put.